parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestRowGroupFilterExactly.java

```diff
@@ -0,0 +1,303 @@
+
+    assertCorrectFiltering(notIn(binaryColumn("name"),
+      Sets.newHashSet(Binary.fromString("miller"), Binary.fromString("anderson"))));
+    assertCorrectFiltering(notIn(binaryColumn("name"),
+      Sets.newHashSet(Binary.fromString("miller"), Binary.fromString("noneExistName"), null)));
+
+    assertCorrectFiltering(lt(longColumn("id"), MAX_ID + 1L));
+    assertCorrectFiltering(lt(longColumn("id"), MAX_ID));
+    assertCorrectFiltering(lt(longColumn("id"), 1234L));
+    assertCorrectFiltering(lt(longColumn("id"), MIN_ID));
+    assertCorrectFiltering(lt(longColumn("id"), MIN_ID - 1L));
+    // for dictionary exactly match less than `miller`
+    assertCorrectFiltering(lt(binaryColumn("name"), Binary.fromString("ailler")));
+    assertCorrectFiltering(lt(binaryColumn("name"), Binary.fromString("miller")));
+
+    assertCorrectFiltering(ltEq(longColumn("id"), MAX_ID + 1L));
+    assertCorrectFiltering(ltEq(longColumn("id"), MAX_ID));
+    assertCorrectFiltering(ltEq(longColumn("id"), 1234L));
+    assertCorrectFiltering(ltEq(longColumn("id"), MIN_ID));
+    assertCorrectFiltering(ltEq(longColumn("id"), MIN_ID - 1L));
+
+    assertCorrectFiltering(gt(longColumn("id"), MAX_ID + 1L));
+    assertCorrectFiltering(gt(longColumn("id"), MAX_ID));
+    assertCorrectFiltering(gt(longColumn("id"), 1234L));
+    assertCorrectFiltering(gt(longColumn("id"), MIN_ID));
+    assertCorrectFiltering(gt(longColumn("id"), MIN_ID - 1L));
+
+    assertCorrectFiltering(gtEq(longColumn("id"), MAX_ID + 1L));
+    assertCorrectFiltering(gtEq(longColumn("id"), MAX_ID));
+    assertCorrectFiltering(gtEq(longColumn("id"), 1234L));
+    assertCorrectFiltering(gtEq(longColumn("id"), MIN_ID));
+    assertCorrectFiltering(gtEq(longColumn("id"), MIN_ID - 1L));
+
+    assertCorrectFiltering(and(eq(binaryColumn("name"), Binary.fromString("noneExistName")),
+      lt(longColumn("id"), -99L)));
+    assertCorrectFiltering(and(eq(binaryColumn("name"), Binary.fromString("miller")),
+      lt(longColumn("id"), 1234L)));
+    assertCorrectFiltering(and(eq(binaryColumn("name"), Binary.fromString("noneExistName")),
+      lt(longColumn("id"), 1234L)));
+
+    assertCorrectFiltering(or(eq(binaryColumn("name"), Binary.fromString("noneExistName")),
+      lt(longColumn("id"), -99L)));
+    assertCorrectFiltering(or(eq(binaryColumn("name"), Binary.fromString("miller")),
+      lt(longColumn("id"), 1234L)));
+    assertCorrectFiltering(or(eq(binaryColumn("name"), Binary.fromString("noneExistName")),
+      lt(longColumn("id"), 1234L)));
+  }
+
+  private void assertCorrectFiltering(FilterPredicate filter) throws IOException {
+    ParquetReadOptions readOptions = ParquetReadOptions.builder()
+      .withRecordFilter(FilterCompat.get(filter)).build();
+
+    // simulate the previous behavior, only skip other filters when predicate is BLOCK_CANNOT_MATCH
+    testEvaluation.setTestExactPredicate(Collections.singletonList(BLOCK_CANNOT_MATCH));
```

Comment:
simulate the previous behavior, only skip other filters when predicate is BLOCK_CANNOT_MATCH